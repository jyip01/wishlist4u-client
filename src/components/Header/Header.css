  .Header {
    height: 40px;
    display: flex;
    align-items: normal;
    justify-content: space-between;
    margin-bottom: 60px;
  }
  
  .Header h1 {
    margin: 0;
  }
  
  .Header h1 a {
    text-decoration: none;
    font-weight: bold;
    text-decoration: underline cornflowerblue;
    text-decoration-style: wavy;
    text-shadow: 0 0 6px #FF0000, 0 0 10px cornflowerblue;
  }
  
  .Header a {
    color:cornsilk;
  }

  .Header h1 a:hover {
    color: cornflowerblue;
    font-size: 37px;
  }
  
  .Header__not-logged-in {
    display: flex;
    flex-direction: column;
    font-weight: bold;
    text-shadow: 0 0 10px black;
  }

  .Header__not-logged-in a:hover {
    color: cornflowerblue;
    font-size: 22px;
    text-shadow: 0 0 6px #FF0000, 0 0 10px cornflowerblue;
  }
  
  .Header__not-logged-in .Hyph {
    display: none;
  }

  .Header__logged-in {
    display: flex;
    flex-direction: column;
    font-weight: bold;
    text-shadow: 0 0 10px black;
  }

  .Header__logged-in a:hover {
    color: cornflowerblue;
    font-size: 22px;
    text-shadow: 0 0 6px #FF0000, 0 0 10px cornflowerblue;
  }
  
  @media screen and (min-width: 450px) {
    .Header {
      align-items: baseline;
    }

    .Header h1 {
      font-size: 35px;
    }
  
    .Header__not-logged-in {
      flex-direction: row;
      font-size: 20px;
    }
  
    .Header__not-logged-in .Hyph {
      display: initial;
    }
  }