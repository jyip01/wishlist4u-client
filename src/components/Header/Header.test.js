import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom'
import Header from './Header';
import renderer from 'react-test-renderer';

describe('Index Rendering', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
    
        ReactDOM.render(<BrowserRouter><Header/></BrowserRouter>,div);
    
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it('renders the UI as expected', () => {
        const tree = renderer.create(<BrowserRouter><Header/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})