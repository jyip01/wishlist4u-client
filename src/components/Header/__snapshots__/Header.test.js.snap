// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Rendering renders the UI as expected 1`] = `
<nav
  className="Header"
>
  <h1>
    <a
      href="/"
      onClick={[Function]}
    >
       WishList4U 
    </a>
  </h1>
  <div
    className="Header__not-logged-in"
  >
    <a
      href="/register"
      onClick={[Function]}
    >
       Create an account 
    </a>
    <span
      className="Hyph"
    >
         -   
    </span>
    <a
      href="/login"
      onClick={[Function]}
    >
       Log in 
    </a>
  </div>
</nav>
`;
